programs/client/Client.cpp

```diff
@@ -1234,6 +1236,160 @@ void Client::processConfig()
     client_info.quota_key = config().getString("quota_key", "");
 }
 
+void Client::readArguments(
```

Comment:
@taiyang-li could you please explain, why we extracted this method as ClientBase abstraction ?

Reply:
@kitaisreal Only in clickhouse-client we need to distinguish whether an option is in external group or not, because `options_description.external_description` is not empty. But in clickhouse-local, we don't have to consider it. So implmentation for clickhouse-client and clickhouse-local seperately is necessary.

Reply:
@kitaisreal do you think it is ok ?

Reply:
Yes, totally ok.